networkx/algorithms/shortest_paths/unweighted.py

```diff
@@ -30,30 +30,27 @@ def single_source_shortest_path_length(G, source, cutoff=None):
        Starting node for path
 
     cutoff : integer, optional
-        Depth to stop the search. Only paths of length <= cutoff are returned.
+        Depth to stop the search. Only paths of length <= `cutoff` are returned.
 
     Returns
     -------
     lengths : dict
-        Dict keyed by node to shortest path length to source.
+        Dict keyed by node to shortest path length to `source`.
 
     Examples
     --------
     >>> G = nx.path_graph(5)
-    >>> length = nx.single_source_shortest_path_length(G, 0)
-    >>> length[4]
-    4
-    >>> for node in length:
-    ...     print(f"{node}: {length[node]}")
-    0: 0
-    1: 1
-    2: 2
-    3: 3
-    4: 4
+    >>> nx.single_source_shortest_path_length(G, 0)
+    {0: 0, 1: 1, 2: 2, 3: 3, 4: 4}
 
     See Also
     --------
-    shortest_path_length
+    :func:`~networkx.algorithms.shortest_paths.generic.shortest_path_length` :
+       Shortest path length with specifiable source, target, and weight.
+    :func:`~networkx.algorithms.shortest_paths.weighted.single_source_dijkstra_path_length` :
+       Shortest weighted path length from source with Dijkstra algorithm.
+    :func:`~networkx.algorithms.shortest_paths.weighted.single_source_bellman_ford_path_length` :
+       Shorttest weighted path length from source with Bellman-Ford algorithm.
```

Comment:
I thought the See Also section automagically made links. Is that no true for this case? Or in general now?

Also, here's a typo fix.
```suggestion
       Shortest weighted path length from source with Bellman-Ford algorithm.
```

Reply:
> I thought the See Also section automagically made links. Is that no true for this case? Or in general now?

There's a catch - it only does so functions that are defined in the same module. For example, even though `shortest_path_length` is obviously related, because it is defined in `networkx.algorithms.shortest_paths.generic` instead of `unweighted`, the link [doesn't work](https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.shortest_paths.unweighted.single_source_shortest_path_length.html#single-source-shortest-path-length).

Reply:
Can we help out sphinx by importing the "See Also" functions into the module? They wouldn't be in `__all__` so I think the namespace wouldn't get polluted beyond that module. 

I'm worried that these links are fairly hard to read in text form. Also, in the text form, they emphasize (due to length) the location of the source in the source tree rather than the name of the function. It is a depth-4, 82-char pointer to the function.   Hmm... Could sphinx use the namespace beyond the current module to look up the See Also info? There must be a reason it uses the source tree instead.

What do you think of importing the see-also functions into each module? Are there downsides to doing that?

Reply:
> Can we help out sphinx by importing the "See Also" functions into the module? They wouldn't be in __all__ so I think the namespace wouldn't get polluted beyond that module.

I'm not sure how this actually works in practice. I suspect that this wouldn't help, as my feeling is that the reference mapping (e.g. `objects.inv`) that sphinx builds is keyed by where the docstrings are found in the source; i.e. this depends only on source traversal rather than import/run-time structure.

I don't know this for sure though, so I will experiment! Another thing which may work is injecting `.. currentmodule::` directives in the see-also section. It doesn't help with aesthetics much, but it would at least eliminate the "explicit-path-in-plain-text" problem. To be clear, I'm not sure this would work either - I'll add it to the experiment list. Ultimately, if we *can* find a solution like this (and one doesn't already exist in `autodoc` that I"m just unaware of), then it'd definitely IMO be worth upstreaming into numpydoc or some other sphinx extension to handle this automatically.

Reply:
Ok, I've tried a few things: the `.. currentmodule` idea is DOA, it breaks the `See Also` parsing.

However, one thing that *does* work is to use an `:any:` role (see 3a9d2ae). This is closer to the intended behavior, i.e. link to the relevant docstring without all of the path cruft. It's not *perfect* because having to explicitly specify the `:any:` role is still a visual wart in the raw docstring, though it's much better than the `:func:` role requiring the full path in the source tree.

This feels like something that could be made configurable in numpydoc, i.e. allow users to configure a default role for see-also entries.